lib/SILGen/SILGenPattern.cpp

```diff
@@ -1923,11 +1923,18 @@ void PatternMatchEmission::emitEnumElementObjectDispatch(
 
         // TODO: If we have something that is not loadable
         if (eltTL->isLoadable()) {
+          // Mark this load as "unenforced" since loading the boxed payload does
+          // not correspond to a formal lvalue access. The box is only immutably
+          // shared across multiple enum lvalues.
+          UnenforcedAccess access;
+          SILValue accessAddress =
+            access.beginAccess(SGF, loc, boxedValue, SILAccessKind::Read);
           // The boxed value may be shared, so we need to load the value at +0
           // to make sure that we copy if we try to use it outside of the switch
           // statement itself.
-          boxedValue = SGF.B.createLoadBorrow(loc, boxedValue);
+          boxedValue = SGF.B.createLoadBorrow(loc, accessAddress);
```

Comment:
I think you need to check here like in the other place if the scope closes immediately (in which case you use a copy, no?)